bolt/lib/Passes/BinaryPasses.cpp

```diff
@@ -1557,15 +1557,28 @@ Error PrintProgramStats::runOnFunctions(BinaryContext &BC) {
       llvm::stable_sort(Functions,
                         [Ascending, &Stats](const BinaryFunction *A,
                                             const BinaryFunction *B) {
-                          return Ascending ? Stats.at(A) < Stats.at(B)
-                                           : Stats.at(B) < Stats.at(A);
+                          auto StatsItr = Stats.find(A);
```

Comment:
Can you please try to factor out the common code into a functor/lambda that would accept a comparison operator to be used? (i.e. operator< in one case and lessThan in another).